sp800-63c/sec4_fal.md

```diff
@@ -0,0 +1,50 @@
+
+Regardless of what is requested or required by the protocol, the FAL in use is easily detected by the RP by observing the nature of the assertion as it is presented as part of the federation protocol. Therefore, the RP is responsible for determining which FALs it is willing to accept for a given authentication transaction and ensuring that the transaction meets the requirements of that FAL.
+
+If the RP is using a front-channel presentation mechanism as defined in [Section 7.2](#front-channel) (e.g., the OpenID Connect Implicit Client profile or the SAML Web SSO profile), it SHOULD require FAL2 or greater in order to protect the information in the assertion from the browser or other parties in the transaction.
+
+### 4.1. Key Management
+
+At any FAL, the IdP SHALL ensure that an RP is unable to impersonate the IdP at another RP by protecting the assertion with a signature and key using approved cryptography. If the assertion is protected by a digital signature using an asymmetric key, the IdP MAY use the same public and private key pair to sign assertions to multiple RPs. The IdP MAY publish its public key in a verifiable fashion, such as at an HTTPS-protected URL at a well-known location. If the assertion is protected by a MAC using a shared key, the IdP SHALL use a different shared key for each RP.
+
+#### 4.2. <a name="runtime-decisions"></a>Runtime Decisions
+
+The fact that parties have federated SHALL NOT be interpreted as permission to pass information. The decision of whether an authentication can occur or attributes are passed can be determined by the use of a whitelist, a blacklist, or a runtime decision by an authorized party.
+
+IdPs MAY establish whitelists of RPs that are authorized to receive authentication and attributes from the IdP, without runtime decision from the subscriber. IdPs MAY also establish blacklists of RPs who are not authorized to receive authentication or attributes from the IdP, even when requested by the subscriber. Both whitelists and blacklists identify RPs by their domain or other sufficiently unique identifier, depending on the federation protocol in use. Every RP that is not on a whitelist or a blacklist SHALL be placed by default in a gray area where runtime authorization decisions will be made by an authorized party, usually the subscriber.
+
+RPs MAY establish whitelists of IdPs that are authorized to send authentication and attributes to the RP, without runtime decision from the subscriber. RPs MAY also establish blacklists of IdPs who are not authorized to send authentication or attributes to the RP, even when requested by the subscriber. Both whitelists and blacklists identify IdPs by their domain or other sufficiently unique identifier, depending on the federation protocol in use. Every IdP that is not on a whitelist or a blacklist SHALL be placed by default in a gray area where runtime authorization decisions will be made by an authorized party, usually the subscriber.
```

Comment:
RP decisions are all about what they request and accept, not about what is sent. I can make the wording clearer.